Improved the table layout

LennartH opened the pull request:
My attempt to improve the table layout for issue #6 that looks like [this](http://imgur.com/a/kmm5U).

I replaced the values array with a dictionary that contains the best results for every formation, the overall best result and the overall second best result. This dictionary is filled by traversing over all infos before they are formatted as a table. This way the infos can be enriched with flags that show which best result they are. This meta data is used upon the table construction, to decide how the different components are formatted.

The code to find the best results is really ugly, but I didn't know how to do it better and I didn't want to change the domain structure. Maybe you know a better way to do this.

Commit:
Improved the table layout

Grimy commented:
Thanks for the PR, it looks great! I really like the new stance headers. However:

* It gives incorrect advice. On your screenshot, the first line says “You should farm on z169 in D”, but looking at the table, you can see that it’s actually more efficient to farm z166 in S.

* It gives gramatically incorrect advice in some cases. With 1e30 attack, world zone 99 and S unlocked, it says “You should farm on z99 **in S in S** or z99. They’re equally efficient.” (emphasis mine).

* The line “It’s 3.6% more efficient than z171.” should specify the stance, as in “It’s 3.6% more efficient than z171 **in D**.” (This seems to be linked to the previous point; it could have the same root cause).

* Before z70, the `X` stance indicator should not be displayed, to avoid confusing players who never unlocked stances.

* `blue.css` is the common stylesheet for all HTML pages in the repo. Those modifications are specific to zFarm, so they should go in the `<style>` tag in `zfarm.html`. Alternatively, you could create a `zfarm.css` file.

I’ll merge if you can fix these issues.

Commit:
Move zFarm specific style to zfarm.html

Commit:
Fix issues with calculating the best results

Commit:
Hide the stance for zones less than 60 and fix NPE

LennartH commented:
Thanks for testing my previous attempt. I have to be honest that I didn't test very thoroughly, because I thought/hoped that you know better how to do it, since you know where to look and what the corner cases are. But how did I not see that my example is just plain wrong.

It took me a while to get it working, but I think that I fixed the issues. The most significant change is that I changed the structure of  the results from a list to a map like structure:

```
{
  'z155': {
    loot: 123,
    stats: {
      'D': {speed: 1.8, value: 221},
      'S': {speed: 1.5, value: 369}
    }
  },
  'z166': ...
}
```

This may be a bit overcomplicated for JavaScript, but that's the way I would have implemented this structure in java.

Another change is that I moved the collecting and marking of the best zones and stances to the calculation of the results in zfarm.js. This looked like a better place to me.

I also think that the stance should be displayed from z60 on, since the H stance is unlocked at that point. But this can be easily changed by editing the check that sets the value of `displayStance` in zfarm.html.

Here are some [example images](http://imgur.com/a/7esBC).

Grimy commented:
Nice work! I didn’t find any issue when testing. I’m also glad you refactored that `I’m really sorry` code.

I merged it in, thanks for contributing!
